.github/workflows/build-binaries.yml

```diff
@@ -0,0 +1,105 @@
+name: Build Binaries
+
+on:
+  workflow_call:
+    inputs:
+      tag_name:
+        description: "Tag name for release builds"
+        required: false
+        type: string
+      is_release:
+        description: "Whether this is a release build"
+        required: true
+        type: boolean
+        default: false
+    outputs:
+      linux_x86_64:
+        description: "Linux x86_64 binary path"
+        value: ${{ jobs.build.outputs.linux_x86_64 }}
+      linux_arm64:
+        description: "Linux ARM64 binary path"
+        value: ${{ jobs.build.outputs.linux_arm64 }}
+      macos_x86_64:
+        description: "macOS x86_64 binary path"
+        value: ${{ jobs.build.outputs.macos_x86_64 }}
+      macos_arm64:
+        description: "macOS ARM64 binary path"
+        value: ${{ jobs.build.outputs.macos_arm64 }}
+      windows_x86_64:
+        description: "Windows x86_64 binary path"
+        value: ${{ jobs.build.outputs.windows_x86_64 }}
+
+jobs:
+  build:
+    name: Build
+    runs-on: ${{ matrix.os }}
+    outputs:
+      linux_x86_64: ${{ steps.build.outputs.linux_x86_64 }}
+      linux_arm64: ${{ steps.build.outputs.linux_arm64 }}
+      macos_x86_64: ${{ steps.build.outputs.macos_x86_64 }}
+      macos_arm64: ${{ steps.build.outputs.macos_arm64 }}
+      windows_x86_64: ${{ steps.build.outputs.windows_x86_64 }}
+    strategy:
+      matrix:
+        include:
+          - name: linux
+            os: ubuntu-22.04 # Use oldest supported non-deprecated version so we link against older glibc version which allows running binary on a wider set of Linux systems
+            path: target/x86_64-unknown-linux-gnu/release/function-runner
+            asset_name: function-runner-x86_64-linux-${{ inputs.is_release && (inputs.tag_name || github.event.release.tag_name) || 'test' }}
```

Comment:
I don't think the `asset_name` is used at all, which means I think we can also remove the `inputs` to this workflow